src/main/java/com/example/exception/BookAlreadyAddedToCartException.java

```diff
@@ -0,0 +1,7 @@
+package com.example.exception;
+
+public class BookAlreadyAddedToCartException extends RuntimeException {
```

Comment:
usually we should use more abstract exception name
Look at comment in shoppingCartServiceImpl - you actually don't need this exception

Reply:
removed